idaes/gas_solid_contactors/properties/oxygen_iron_OC_oxidation/hetero_reactions.py

```diff
@@ -0,0 +1,517 @@
+                        "object for the gas phase.",
+            domain=is_physical_parameter_block))
+    CONFIG.declare("solid_property_package", ConfigValue(
+            description="Reference to associated PropertyPackageParameter "
+                        "object for the solid phase.",
+            domain=is_physical_parameter_block))
+    CONFIG.declare("default_arguments", ConfigBlock(
+            description="Default arguments to use with Property Package",
+            implicit=True))
+
+    def build(self):
+        '''
+        Callable method for Block construction.
+        '''
+        super(ReactionParameterBlock, self).build()
+
+        self._reaction_block_class = ReactionBlock
+
+        # Create Phase objects
+        self.Vap = VaporPhase()
+        self.Sol = SolidPhase()
+
+        # Create Component objects
+        self.N2 = Component()
+        self.O2 = Component()
+        self.CO2 = Component()
+        self.H2O = Component()
+        self.Fe2O3 = Component()
+        self.Fe3O4 = Component()
+        self.Al2O3 = Component()
+
+        # Component list subsets
+        self.gas_component_list = Set(initialize=['O2', 'N2', 'CO2', 'H2O'])
+        self.sol_component_list = Set(initialize=['Fe2O3', 'Fe3O4', 'Al2O3'])
+
+        # Reaction Index
+        self.rate_reaction_idx = Set(initialize=["R1"])
+
+        # Gas Constant
+        self.gas_const = Param(within=PositiveReals,
+                               mutable=False,
+                               default=8.314459848e-3,
+                               doc='Gas Constant [kJ/mol.K]')
+
+        # Reaction Stoichiometry
+        self.rate_reaction_stoichiometry = {("R1", "Vap", "O2"): -1,
+                                            ("R1", "Vap", "N2"): 0,
+                                            ("R1", "Vap", "CO2"): 0,
+                                            ("R1", "Vap", "H2O"): 0,
+                                            ("R1", "Sol", "Fe2O3"): 6,
+                                            ("R1", "Sol", "Fe3O4"): -4,
+                                            ("R1", "Sol", "Al2O3"): 0}
+
+        # Reaction stoichiometric coefficient
+        self.rxn_stoich_coeff = Param(self.rate_reaction_idx,
+                                      default=4,
+                                      mutable=True,
+                                      doc='Reaction stoichiometric'
+                                      'coefficient [-]')
+
+        # Standard Heat of Reaction - kJ/mol_rxn
+        dh_rxn_dict = {"R1": -469.4432}
+        self.dh_rxn = Param(self.rate_reaction_idx,
+                            initialize=dh_rxn_dict,
+                            doc="Heat of reaction [kJ/mol]")
+
+        # Smoothing factor
+        self.eps = Param(mutable=True,
+                         default=1e-8,
+                         doc='Smoothing Factor')
+        # Reaction rate scale factor
+        self._scale_factor_rxn = Param(mutable=True,
```

Comment:
A note for the future, but we should also get the IDAES scaling tools supported here so that we can remove these (semi-)hard-coded scaling factors.